internal/acctest/service/base/rate_limit_configuration.go

```diff
@@ -0,0 +1,88 @@
+// Copyright © 2025 Ping Identity Corporation
+
+package base
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/patrickcping/pingone-go-sdk-v2/management"
+	"github.com/pingidentity/terraform-provider-pingone/internal/acctest"
+	"github.com/pingidentity/terraform-provider-pingone/internal/acctest/legacysdk"
+)
+
+func RateLimitConfiguration_CheckDestroy(s *terraform.State) error {
+	var ctx = context.Background()
+
+	p1Client, err := legacysdk.TestClient(ctx)
+
+	if err != nil {
+		return err
+	}
+
+	apiClient := p1Client.API.ManagementAPIClient
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "pingone_rate_limit_configuration" {
+			continue
+		}
+
+		shouldContinue, err := legacysdk.CheckParentEnvironmentDestroy(ctx, apiClient, rs.Primary.Attributes["environment_id"])
+		if err != nil {
+			return err
+		}
+
+		if shouldContinue {
+			continue
+		}
+
+		_, r, err := apiClient.RateLimitingApi.ReadOneRateLimitConfiguration(ctx, rs.Primary.Attributes["environment_id"], rs.Primary.ID).Execute()
+
+		shouldContinue, err = acctest.CheckForResourceDestroy(r, err)
+		if err != nil {
+			return err
+		}
+
+		if shouldContinue {
+			continue
+		}
+
+		return fmt.Errorf("PingOne Rate Limit Configuration %s still exists", rs.Primary.ID)
+	}
+
+	return nil
+}
+
+func RateLimitConfiguration_GetIDs(resourceName string, environmentID, resourceID *string) resource.TestCheckFunc {
```

Comment:
It seems like this doesn't do anything unique to rate limits. If we have a common version of this function, we could use that instead, but if this is just the typical model our tests follow that's fine.